go/test/endtoend/multiorch_helpers.go

```diff
@@ -0,0 +1,632 @@
+	env.t.Helper()
+
+	nodes := make([]*nodeInstance, count)
+	for i := range count {
+		node := createEmptyNode(env.t, env.tempDir, env.config.cellName,
+			env.config.shardID, env.config.database, i, env.etcdClientAddr, env.config.stanzaName)
+		nodes[i] = node
+		env.t.Cleanup(func() { cleanupNode(env.t, node) })
+	}
+	env.nodes = nodes
+	env.t.Logf("Created %d nodes with pgctld running", count)
+	return nodes
+}
+
+// setupPgBackRest sets up pgbackrest configuration for all nodes in the environment
+func (env *testEnv) setupPgBackRest() {
+	env.t.Helper()
+	setupPgBackRestForBootstrap(env.t, env.tempDir, env.nodes, env.config.stanzaName)
+}
+
+// registerNodes registers all nodes in the topology
+func (env *testEnv) registerNodes() {
+	env.t.Helper()
+	ctx := env.t.Context()
+
+	for _, node := range env.nodes {
+		pooler := &clustermetadatapb.MultiPooler{
+			Id: &clustermetadatapb.ID{
+				Component: clustermetadatapb.ID_MULTIPOOLER,
+				Cell:      node.cell,
+				Name:      node.name,
+			},
+			Hostname: "localhost",
+			PortMap: map[string]int32{
+				"grpc": int32(node.grpcPort),
+			},
+			Shard:      env.config.shardID,
+			Database:   env.config.database,
+			TableGroup: env.config.tableGroup,
+			Type:       clustermetadatapb.PoolerType_UNKNOWN, // All start with unknown type until bootstrap determines role
+		}
+		err := env.ts.RegisterMultiPooler(ctx, pooler, true /* overwrite */)
+		require.NoError(env.t, err, "Failed to register pooler %s in topology", node.name)
+		env.t.Logf("Registered pooler %s in topology", node.name)
+	}
+}
+
+// startMultiOrch starts multiorch with the configured watch targets
+func (env *testEnv) startMultiOrch() *exec.Cmd {
+	env.t.Helper()
+
+	watchTarget := fmt.Sprintf("%s/%s/%s", env.config.database, env.config.tableGroup, env.config.shardID)
+	cmd := startMultiOrch(env.t, env.tempDir, env.config.cellName, env.etcdClientAddr, []string{watchTarget})
+	env.multiOrchCmd = cmd
+	env.t.Cleanup(func() { terminateProcess(env.t, cmd, "multiorch", 5*time.Second) })
+	return cmd
+}
+
+// createEmptyNode creates a new empty multipooler node with pgctld and multipooler processes
+func createEmptyNode(t *testing.T, baseDir, cell, shard, database string, index int, etcdAddr, pgBackRestStanza string) *nodeInstance {
+	t.Helper()
+
+	name := fmt.Sprintf("node%d", index)
+	dataDir := filepath.Join(baseDir, name)
+	require.NoError(t, os.MkdirAll(dataDir, 0o755))
+
+	// Allocate ports
+	pgctldGrpcPort := utils.GetFreePort(t)
+	pgPort := utils.GetFreePort(t)
+	grpcPort := utils.GetFreePort(t)
+
+	// Start pgctld server
+	logFile := filepath.Join(dataDir, "pgctld.log")
+	pgctldCmd := exec.Command("pgctld", "server",
+		"--pooler-dir", dataDir,
+		"--grpc-port", fmt.Sprintf("%d", pgctldGrpcPort),
+		"--pg-port", fmt.Sprintf("%d", pgPort),
+		"--log-output", logFile)
+
+	// Set MULTIGRES_TESTDATA_DIR for directory-deletion triggered cleanup
+	pgctldCmd.Env = append(os.Environ(),
```

Comment:
FYI, I also may send out a change I have in a branch somewhere to wrap all these exec.Command() calls in a helper so that things like setting this environment variable are more fluid